lib/methods.rb

```diff
@@ -1,80 +1,62 @@
 # frozen_string_literal: true
 
-require_relative "./player.rb"
-
-class GameManager
-  attr_reader :board, :player_one, :player_two, :X, :O
-  def initialize(players, board = (1..9).to_a)
-    @board = board
-    @player_one = Player.new(players)
-    @player_two = Player.new(players)
-  end
-
-  WIN_COMBINATIONS = [
-    [0, 1, 2],
-    [3, 4, 5],
-    [6, 7, 8],
-    [0, 3, 6],
-    [1, 4, 7],
-    [2, 5, 8],
-    [6, 4, 2],
-    [0, 4, 8]
-  ].freeze
-
-  def won?
-    WIN_COMBINATIONS.detect do |combo|
-      @board[combo[0]] == @board[combo[1]] &&
-        @board[combo[1]] == @board[combo[2]] &&
-        space_filled?(combo[0])
-    end
-  end
-
-  def set_input(index, input = :X)
-    @board[index] = input
-  end
-
-  def space_filled?(index)
-    @board[index] == :X || @board[index] == :O
+class Board
+  attr_accessor :game_board, :board_num
+  def initialize
+    @game_board = [" ", " ", " ", " ", " ", " ", " ", " ", " "]
+    @board_num = []
   end
 
-  def board_index(input_index)
-    input_index.to_i - 1
+  def display_board
+    puts "#{@game_board[0]}|#{@game_board[1]}|#{@game_board[2]}\n"\
+     "---------\n"\
+     "#{@game_board[3]}|#{@game_board[4]}|#{@game_board[5]}\n"\
+     "---------\n"\
+     "#{@game_board[6]}|#{@game_board[7]}|#{@game_board[8]}"
   end
 
-  def valid_move?(position)
-    position.between?(0, 8) && !space_filled?(position)
+  def add_to_board(choice, player)
+    @game_board[choice - 1] = player
   end
 
-  def turn_count
-    @board.select { |e| e == :X || e == :O }.size # rubocop:disable Style/MultipleComparison
+  def count_board(player)
+    @board_num = []
+    9.times do |i|
+        @board_num << i if @game_board[i] == player
+        @board_num << nil if @game_board[i] != player
+    end
```

Comment:
[OPTIONAL]
You can achieve this with the ternary operator.

```
@board_num << @game_board[i] == player ? i : nil
```

Reply:
Thanks bro.